File: linked-list/doubly_linked_list.py
class Node:
    def __init__(self):
        self.data = None
        self.nextn = None
        self.prevn = None


X = 10

for i in range(0,X):
    if(i==0):
        temp = Node()
        temp.data = i

        head = temp
        tail = temp
    else:
        temp = Node()
        temp.data = i

        temp.prevn = tail
        tail.nextn = temp

        tail = temp
    

# from head to tail print
current_p = head
while current_p != None:
    print(current_p.data,end= " ")
    current_p = current_p.nextn

print()
#from tail to head printing
current_p = tail
while current_p != None:
    print(current_p.data, end= " ")
    current_p = current_p.prevn

File: linked-list/doubly_linked_list.c
#include<stdio.h>
#include<stdlib.h>

typedef struct node
{
    int data;
    struct node* next;
    struct node* prev;
}node;

node* insert_element(int n);

int main()
{
    int n = 10;

    node* p = insert_element(n);

    while(p!=NULL)
    {
        printf("%d\n",p->data);
        p = p->next;
    }
    
}

node* insert_element(int n)
{
    node* head = NULL;
    node* tail = NULL;
    node* temp = NULL;

    for(int i=0;i<n;i++)
    {
        temp = malloc(sizeof(node));
        if(temp == NULL)
        {
            printf("memory allocation failed!");
            exit(1);
        }

        if(i==0)
        {
            temp->data = i;

            head = temp;
            tail = temp;
            temp->prev = NULL;
            temp->next = NULL;
        }

        else{
            temp->data = i;

            temp->prev = tail;
            temp->next = NULL;

            tail->next = temp;

            tail = temp;
        }
    }

    return head;
}
